src/app/core/public/api/impl/BaseObject.ts

```diff
@@ -37,10 +37,14 @@ export abstract class BaseObjectImpl implements BaseObject {
     }
 
     public set isSelected(val: boolean) {
-        throw new Error("Unimplemented");
+        if (val === true) {
```

Comment:
nit: just do `if (val)` and also remove all the curly braces since these are 1-liner if/else statement blocks